actioncable/lib/action_cable/subscription_adapter/postgresql.rb

```diff
@@ -29,15 +29,20 @@ def shutdown
       end
 
       def with_connection(&block) # :nodoc:
-        ActiveRecord::Base.connection_pool.with_connection do |ar_conn|
-          pg_conn = ar_conn.raw_connection
-
-          unless pg_conn.is_a?(PG::Connection)
-            raise "The Active Record database must be PostgreSQL in order to use the PostgreSQL Action Cable storage adapter"
-          end
+        ar_conn = ActiveRecord::Base.connection_pool.checkout.tap do |conn|
+          # Action Cable is taking ownership over this database connection, and
+          # will perform the necessary cleanup tasks
+          ActiveRecord::Base.connection_pool.remove(conn)
+        end
```

Comment:
AR's "connect and give up control" public API feels like a narrower dependency than trying to steal its connection config and connect for ourselves... and forcing the user to duplicate the details just seems unnecessary.